mmv1/third_party/terraform/services/container/resource_container_cluster.go.tmpl

```diff
@@ -1722,13 +1728,120 @@ func ResourceContainerCluster() *schema.Resource {
 				ConflictsWith: []string{"enable_autopilot"},
 			},
 
+			"control_plane_endpoints_config": {
```

Comment:
Is a user able to adopt this field for `master_authorized_networks_config` and the group of `private_cluster_config` subfields independently, or do they need to modify both all at once?

Reply:
It can migrated incrementally/independently. See expandControlPlaneEndpointsConfig line 5619 where it first checks for d.GetOk("master_authorized_networks_config") when filling in the ip_endpoints_config to the server. This way, it'll always use the new fields in the API request to GKE, but users can migrate their TF config as slowly as they want.

Reply:
Although, the way it's written now, the block of private_cluster_config fields needs to be migrated as one unit. So you can't, for instance, migrating enable_public_endpoints to control_plane_endpoints_config, but leave global_access in private_cluster_config. I'm not sure if we'd need to have additional validation to prevent that, or if it's better to just rewrite it to look for fallbacks for each and every field.

Reply:
> I'm not sure if we'd need to have additional validation to prevent that, or if it's better to just rewrite it to look for fallbacks for each and every field.

Either approach can work, I don't have a strong opinion on which we do as long as we end up in the right final state for any given configuration.

Reply:
OK, I've updated the ConflictsWith attributes to ensure that users can't configure `private_cluster_config` in a way that `expandControlPlaneEndpointsConfig` doesn't handle.